### ISSUE TITLE
Fedora 38 can't use Vulkan backend if `vulkan-validation-layers` is installed

### ISSUE THREAD

**Issue by jimblandy** [gfx-rs/wgpu#3712]
When the `vulkan-validation-layers` package is installed, `wgpu` will refuse to use the Vulkan backend in debug builds.

If you run `vkcube --validate`, that crashes too, so this isn't a `wgpu` problem, but I'm filing this issue just because people are going to run into it in `wgpu`.

There's an upstream bug filed: https://bugzilla.redhat.com/show_bug.cgi?id=2183567

Just for completeness, here's how `wgpu` ends up asking for the validation layer in the first place:

[`wgpu_core::Instance::new`](https://github.com/gfx-rs/wgpu/blob/3432aabab158b2183d4eb2b6f58b74639acad2f3/wgpu-core/src/instance.rs#L74-L77) requests the validation and debug layers:

```
if cfg!(debug_assertions) {
    flags |= hal::InstanceFlags::VALIDATION;
    flags |= hal::InstanceFlags::DEBUG;
}
```

This causes [`<wgpu_hal::vulkan::Instance as wgpu_hal::Instance>::init`](https://github.com/gfx-rs/wgpu/blob/3432aabab158b2183d4eb2b6f58b74639acad2f3/wgpu-hal/src/vulkan/instance.rs#L554-L556) to try to add the validation layer:

```
if desc.flags.contains(crate::InstanceFlags::VALIDATION) {
    layers.push(CStr::from_bytes_with_nul(b"VK_LAYER_KHRONOS_validation\0").unwrap());
}
```

**Comment by jimblandy** [comment]
I cloned `git@github.com:KhronosGroup/Vulkan-ValidationLayers.git`, built it according to the instructions in `BUILD.md`, and now I can run `wgpu` programs with validation:

```
$ cd
$ git clone git@github.com:KhronosGroup/Vulkan-ValidationLayers.git
$ cd Vulkan-ValidationLayers/
$ ./scripts/update_deps.py
$ cmake -S . -B build -D UPDATE_DEPS=ON -D BUILD_WERROR=ON -D BUILD_TESTS=ON -D CMAKE_BUILD_TYPE=Release
$ cmake --build build --config Release -j
$ cd ~/wgpu
$ VK_LAYER_PATH=/home/jimb/Vulkan-ValidationLayers/build/layers cargo run -p wgpu --example water
```

**Comment by jimblandy** [comment]
This has been fixed in `vulkan-validation-layers-1.3.243.0-1.fc38`.
